### PR TITLE
Sitemaps: use Jetpack_PostImages::get_image to get any image in entry, closes #3333

### LINKED ISSUE #3333: Sitemaps: use Jetpack_PostImages for regular sitemaps as well
In 57571c1818a64478459c747063991038db95a2a8, we started using `Jetpack_PostImages` to look for an image to be used in the News sitemap.

I think we should follow the same method for regular sitemaps as well, instead of relying on code that ignores all images that aren't attached to the post:
https://github.com/Automattic/jetpack/blob/3.9.1/modules/sitemaps/sitemaps.php#L284

Reported here:
https://wordpress.org/support/topic/sitemap-and-featured-image?replies=1&view=all

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #3333

### CODE REVIEW

**Comment by jeherve** [comment]
It looks like the previous implementation created multiple image tags if there were multiple images attached to the post. Maybe we should use `get_images` instead of `get_image`, and return all the images we can.

As per Google:
> Each `<url>` tag can contain up to 1,000 `<image:image>` tags.
https://support.google.com/webmasters/answer/178636?hl=en

It might be useful to change `get_images`' parameters and set `from_html` to `false`, since we probably don't want to include images that are hotlinked from another site in our sitemap.

**Comment by eliorivero** [comment]
It actually didn't, only the first one found was included, but it's certainly something we should include so I'll look into this.

File: modules/sitemaps/sitemaps.php [hunk 1]
```diff
@@ -351,28 +319,43 @@ function jetpack_print_sitemap() {
...
+			$methods = array(
+				'from_thumbnail'  => false,
+				'from_slideshow'  => false,
+				'from_gallery'    => false,
+				'from_attachment' => false,
+				'from_html'       => false,
+			);
+			foreach ( $methods as $method => $value ) {
+				$methods[ $method ] = true;
+				$images_collected = Jetpack_PostImages::get_images( $post->ID, $methods );
```
Comment (zinigor) [inline modules/sitemaps/sitemaps.php]: I think you meant `$method` here instead of `$methods`.

**Comment by zinigor** [comment]
Read, found one small problem, tested, looks good after fixing!

File: modules/sitemaps/sitemaps.php (hunk 1 above)
Comment (eliorivero) [inline modules/sitemaps/sitemaps.php]: Just checked it and it's ok, the variable passed to the `get_images` method is `$methods`.

Comment (zinigor) [inline modules/sitemaps/sitemaps.php]: What is the `foreach` for then? If you do `get_images` with the same arguments in each iteration, you will get the same results each time, won't you?

**Comment by samhotchkiss** [comment]
Since this is both a pretty major change and involves string changes, I'd be more comfortable if we included it in 3.10 (as always, I'm open to discussion on it)

**Comment by zinigor** [comment]
In other cases I would too, but this is a fix for a feature we just released as part of 3.9.0, and it looks to be a rather significant improvement. So if we figure out that thing with `$methods`, I feel pretty good about it :)

File: modules/sitemaps/sitemaps.php (hunk 1 above)
Comment (eliorivero) [inline modules/sitemaps/sitemaps.php]: It's exactly for that: to fetch a different image or set of images each time. Check that in first iteration, the `from_thumbnail` item is set to true, the rest is false. After the image is fetch, the item is set to false again. In the second iteration, only `from_slideshow` is set to true. And so on. In this way, we collect all the images that `get_images`method can fetch.

Comment (zinigor) [inline modules/sitemaps/sitemaps.php]: Now I see it, thanks for explaining!

**Comment by zinigor** [comment]
OK, looks good to go!

**Pull request merged**